# AMZ-Driverless
# Copyright (c) 2018 Authors:
#   - Juraj Kabzan <kabzanj@gmail.com>
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.

car:

  inertia:
    m:        190.0   # Weight of the Vehicle [kg]
    m_driver: 0.0     # Weight of the driver  [kg]
    g:        9.81    # Graviti force         [m/s^2]
    I_z:      110    # Inertial force I_zz 

  kinematics:
    l: 1.53           # Vehicle Length [m]
    b_F: 1.22          # From COG to front axle [m]
    b_R: 1.22         # From COG to rear axle [m]
    w_front: 0.5    # Percentage of weight front
    h_cg: 0.262       # Heights of COG [m]
    l_width: 3.0     # Width of the vehicle body

    # l_F = l * (1 - w_front);  # Front cg lever arm [m]
    # l_R = l * w_front;  # Rear  cg lever arm [m]

  # Magic Formula:
  # y=D*sin(C*atan(B*(1-E)*x+E*atan(B*x)))
  tire:
    tire_coefficient: 1.0
    B: 12.56
    C: -1.38
    D: 1.60
    E: -0.58

  aero:
    C_Down: # F_Downforce = C_Downforce*v_x^2; C_Downforce = a*b*c
      a: 1.22
      b: 2.6
      c: 0.6
    C_drag: # F_Drag = C_Drag*v_x^2; C_drag = a*b*c
      a: 0.7
      b: 1.0
      c: 1.0

  drivetrain:
    Cm1: 5000
    Cr0: 180
    inertia: 0.4 # wheel_plus_packaging
    r_dyn: 0.231
    nm_wheels: 4
    
  torque_vectoring:
    K_FFW: 2.1
    K_p: 500 
    shrinkage: 0.8
    K_stability: 0.004


